lib/nostrum/struct/message.ex

```diff
@@ -163,31 +224,39 @@ defmodule Nostrum.Struct.Message do
     new =
       map
       |> Map.new(fn {k, v} -> {Util.maybe_to_atom(k), v} end)
-      |> Map.update(:id, nil, &Util.cast(&1, Snowflake))
-      |> Map.update(:guild_id, nil, &Util.cast(&1, Snowflake))
-      |> Map.update(:channel_id, nil, &Util.cast(&1, Snowflake))
-      |> Map.update(:author, nil, &Util.cast(&1, {:struct, User}))
-      |> Map.update(:mentions, nil, &Util.cast(&1, {:list, {:struct, User}}))
-      |> Map.update(:mention_roles, nil, &Util.cast(&1, {:list, Snowflake}))
-      |> Map.update(:attachments, nil, &Util.cast(&1, {:list, {:struct, Attachment}}))
-      |> Map.update(:embeds, nil, &Util.cast(&1, {:list, {:struct, Embed}}))
-      |> Map.update(:reactions, nil, &Util.cast(&1, {:list, {:struct, Reaction}}))
-      |> Map.update(:nonce, nil, &Util.cast(&1, Snowflake))
-      |> Map.update(:webhook_id, nil, &Util.cast(&1, Snowflake))
       |> Map.update(:activity, nil, &Util.cast(&1, {:struct, Activity}))
+      |> Map.update(:application_id, nil, &Util.cast(&1, Snowflake))
       |> Map.update(:application, nil, &Util.cast(&1, {:struct, Application}))
+      |> Map.update(:attachments, nil, &Util.cast(&1, {:list, {:struct, Attachment}}))
+      |> Map.update(:author, nil, &Util.cast(&1, {:struct, User}))
+      |> Map.update(:channel_id, nil, &Util.cast(&1, Snowflake))
+      |> Map.update(:components, nil, &Util.cast(&1, {:list, {:struct, Component}}))
+      |> Map.update(:embeds, nil, &Util.cast(&1, {:list, {:struct, Embed}}))
+      |> Map.update(:guild_id, nil, &Util.cast(&1, Snowflake))
+      |> Map.update(:id, nil, &Util.cast(&1, Snowflake))
+      |> Map.update(:interaction, nil, &Util.cast(&1, {:struct, Channel}))
       |> Map.update(:member, nil, &Util.cast(&1, {:struct, Member}))
+      |> Map.update(:mention_channels, nil, &Util.cast(&1, {:list, {:struct, Channel}}))
+      |> Map.update(:mention_roles, nil, &Util.cast(&1, {:list, Snowflake}))
+      |> Map.update(:mentions, nil, &Util.cast(&1, {:list, {:struct, User}}))
       |> Map.update(:message_reference, nil, &Util.cast(&1, {:struct, Reference}))
+      |> Map.update(:nonce, nil, &Util.cast(&1, Snowflake))
+      |> Map.update(:reactions, nil, &Util.cast(&1, {:list, {:struct, Reaction}}))
       |> Map.update(:referenced_message, nil, &Util.cast(&1, {:struct, __MODULE__}))
+      |> Map.update(:sticker_items, nil, &Util.cast(&1, {:list, {:struct, Stickers}}))
```

Comment:
This should probably be `Sticker`:

```suggestion
      |> Map.update(:sticker_items, nil, &Util.cast(&1, {:list, {:struct, Sticker}}))
```